### PR TITLE
Add nanoseconds to instant protocol

### PR DESCRIPTION
**Pull request by ktoso**
**Motivation:**

we have nanoseconds precision on most platforms so lets expose it.

resolves https://github.com/apple/swift-distributed-tracing/issues/105

### CODE REVIEW

**Comment by ktoso** [comment]
Also resolves https://github.com/apple/swift-distributed-tracing/issues/103 by removing unused protocol

File: Sources/Tracing/TracingTime.swift [hunk 1]
```diff
@@ -21,26 +21,17 @@ import Darwin
...
-    static func *= (_ lhs: inout Self, _ rhs: Int)
-
-    static func / (_ lhs: Self, _ rhs: Self) -> Double
-}
-
-extension SwiftDistributedTracingDurationProtocol {
-    public static func /= (_ lhs: inout Self, _ rhs: Int) {
-        lhs = lhs / rhs
-    }
+public protocol TracerInstantProtocol: Comparable, Hashable, Sendable {
```
Comment (stevapple) [inline Sources/Tracing/TracingTime.swift:24]: This is one of the two last places where we have the `Protocol` suffix. Do we want to get rid of it entirely?

Comment (ktoso) [reply Sources/Tracing/TracingTime.swift:24]: Yeah I think we can :+1:

I'll do this in this PR actually. Implementations tend to be called Timestamp or something like that after all

**Review by Lukasa — APPROVED** [review]

**Review by fabianfett — APPROVED** [review]

**Review by slashmo — APPROVED** [review]

**Comment by ktoso** [comment]
Sweet, thanks folks; moving on then

**Pull request merged**